### PR TITLE
Fix resolving integer divisions

### LINKED ISSUE #4516: Sympy Division by Integer in Exponent or Radians throws and error
**Description of the issue**
When using sympy to define a circuit, a ValueError exception might be thrown by NumPy
(ValueError: Integers to negative integer powers are not allowed.)

This happens when there is some division in the exponent by some integer and an attempt to resolve the parameters of the gate/circuit is made.

A simple example of this issue is given below.

**How to reproduce the issue**

```

import cirq
import sympy

Rx_gate = cirq.rx(rads=sympy.Symbol('a')/sympy.Symbol('b'))
cirq.resolve_parameters(Rx_gate,{'a':1, 'b':2})

```

<details>

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-101-ef0861896379> in <module>
----> 1 cirq.resolve_parameters(Rx_1,{'a':a, 'b':b, 't':t, 'm':m})

~/anaconda3/envs/quantum/lib/python3.8/site-packages/cirq/protocols/resolve_parameters.py in resolve_parameters(val, param_resolver, recursive)
    182         result = NotImplemented
    183     else:
--> 184         result = getter(param_resolver, recursive)
    185 
    186     if result is not NotImplemented:

~/anaconda3/envs/quantum/lib/python3.8/site-packages/cirq/ops/eigen_gate.py in _resolve_parameters_(self, resolver, recursive)
    354 
    355     def _resolve_parameters_(self, resolver: 'cirq.ParamResolver', recursive: bool) -> 'EigenGate':
--> 356         return self._with_exponent(exponent=resolver.value_of(self._exponent, recursive))
    357 
    358     def _equal_up_to_global_phase_(self, other, atol):

~/anaconda3/envs/quantum/lib/python3.8/site-packages/cirq/study/resolver.py in value_of(self, value, recursive)
    141             product = self.value_of(value.args[0], recursive)
    142             for factor in value.args[1:]:
--> 143                 product *= self.value_of(factor, recursive)
    144             return product
    145         if isinstance(value, sympy.Pow) and len(value.args) == 2:

~/anaconda3/envs/quantum/lib/python3.8/site-packages/cirq/study/resolver.py in value_of(self, value, recursive)
    144             return product
    145         if isinstance(value, sympy.Pow) and len(value.args) == 2:
--> 146             return np.power(
    147                 self.value_of(value.args[0], recursive), self.value_of(value.args[1], recursive)
    148             )

ValueError: Integers to negative integer powers are not allowed.

</details>

**Cirq version**
0.11.1

### PR DESCRIPTION
**Pull request by eendebakpt**
The cirq parameter resolver internally used `np.power` to evaluate `a/b`. This fails for integer values of `a` and `b`. This PR casts to a float, so that the following is allowed:
```
import cirq
import sympy

Rx_gate = cirq.rx(rads=sympy.Symbol('a')/sympy.Symbol('b'))
cirq.resolve_parameters(Rx_gate,{'a':1, 'b':2})
```
Fixes #4516

### CODE REVIEW

File: cirq-core/cirq/study/resolver.py [hunk 1]
```diff
@@ -143,9 +143,10 @@ def value_of(
                 product *= self.value_of(factor, recursive)
             return product
         if isinstance(value, sympy.Pow) and len(value.args) == 2:
-            return np.power(
-                self.value_of(value.args[0], recursive), self.value_of(value.args[1], recursive)
-            )
+            base = self.value_of(value.args[0], recursive)
+            exponent = self.value_of(value.args[1], recursive)
+
```
Comment (MichaelBroughton) [inline cirq-core/cirq/study/resolver.py:148]: Nit: don't need a blank space here.

File: cirq-core/cirq/study/resolver.py [hunk 2]
```diff
@@ -143,9 +143,10 @@ def value_of(
...
+            return np.power(float(base), exponent)
```
Comment (MichaelBroughton) [inline cirq-core/cirq/study/resolver.py:149]: Could we use `np.float_power` instead ?

**Review by MichaelBroughton — CHANGES_REQUESTED** [review]

File: cirq-core/cirq/study/resolver.py (hunk 2 above)
Comment (eendebakpt) [reply cirq-core/cirq/study/resolver.py]: @MichaelBroughton Done!

**Review by MichaelBroughton — APPROVED** [review]
LGTM

**Comment by MichaelBroughton** [comment]
Oops, forgot to approve the CI runs and mistook that for passing all the tests. @eendebakpt do you want to take a crack at trying to fix some of these CI failures ?

**Comment by peendebak** [comment]
> Oops, forgot to approve the CI runs and mistook that for passing all the tests. @eendebakpt do you want to take a crack at trying to fix some of these CI failures ?

@MichaelBroughton One issue was with the `np.float_power` not being able to handle sympy objects. I updated the code for this, but there might be a few more cases to handle (I am having some problems running tests locally).

**Comment by peendebak** [comment]
@MichaelBroughton Could you approve CI once more? Thanks

File: cirq-core/cirq/study/resolver.py [hunk 3]
```diff
@@ -143,9 +143,12 @@ def value_of(
...
             return product
         if isinstance(value, sympy.Pow) and len(value.args) == 2:
-            return np.power(
-                self.value_of(value.args[0], recursive), self.value_of(value.args[1], recursive)
-            )
+            base = self.value_of(value.args[0], recursive)
+            exponent = self.value_of(value.args[1], recursive)
+            if isinstance(base, numbers.Number):
+                return np.float_power(base, exponent)
+            else:
```
Comment (MichaelBroughton) [inline cirq-core/cirq/study/resolver.py:150]: Nit: do we need this else line ?

**Review by MichaelBroughton — APPROVED** [review]
LGTM with one last nit.

File: cirq-core/cirq/study/resolver.py (hunk 3 above)
Comment (peendebak) [reply cirq-core/cirq/study/resolver.py]: We don't need it, just a matter of taste. I removed it!

**Pull request merged**